contracts/staking/Staking.sol

```diff
@@ -1010,9 +1010,21 @@ contract Staking is StakingV2Storage, GraphUpgradeable, IStaking, Multicall {
             // When allocation is closed redirect funds to the rebate pool
             // This way we can keep collecting tokens even after the allocation is closed and
             // before it gets to the finalized state.
+            // If the allocation didn't collect anything before, also add
+            // effectiveAllocation to the rebatePool
             if (allocState == AllocationState.Closed) {
                 Rebates.Pool storage rebatePool = rebates[alloc.closedAtEpoch];
-                rebatePool.fees = rebatePool.fees.add(queryFees);
+                if (queryFees == alloc.collectedFees) {
```

Comment:
Couldn't this part just do `if (!rebatePool.exists())` instead of `if (queryFees == alloc.collectedFees)` ? not sure we need the extra if

I'm thinking if we can wrap this functionality into addToPool

Reply:
I added the if in case `rebatePool.exists()` was quite gas intensive, but I assume that yes, it could be directly checked, the only thing is that the qfee check is also important to ratify whether we should add only the qfees part, or both the qfees and effective allocation (mirroring the behaviour of allocation closed)

Reply:
`.exists()` is one SLOAD and for the `alloc.collectedFees` we are using one too. I'll think if I can find a more compact way to do it, but the logic makes sense.

Another thing that I want to check is that nothing goes wrong with the `unclaimedAllocationsCount` in the zero fee cases

Reply:
We could use a local variable above (line 1008):

```
            // Add funds to the allocation
            allocCollectedFees = alloc.collectedFees.add(queryFees);
            alloc.collectedFees = allocCollectedFees
```

and then use `allocCollectedFees == queryFees` - that should save us the SLOAD?

Reply:
(unless the compiler / optimizer do this already, probably worth checking with a gas test)